src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateDuplicate = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
```

Comment:
You can make a destructuring to get all the needed props of action(type, extraData, keysToRemove)